autopush/websocket.py

```diff
@@ -412,13 +421,35 @@ def process_hello(self, data):
         uaid = data.get("uaid")
         _, uaid = validate_uaid(uaid)
         self.uaid = uaid
-
+        # Check for the special UDP wakeup commands
+        wakeup_host = data.get("wakeup_host")
+        mobilenetwork = data.get("mobilenetwork")
+        self.udp = None
+        # If this connection uses the UDP wakeup mechanism, add it.
+        try:
+            # Normalize the UDP wake info to a single object.
+            udp_data = dict(ip=wakeup_host.get("ip"),
+                            port=wakeup_host.get("port"),
+                            mcc=mobilenetwork.get("mcc", ''),
+                            mnc=mobilenetwork.get("mnc", ''),
+                            netid=mobilenetwork.get("netid", ''))
+            self.udp = dict(data=urlencode(udp_data),
+                            timeout=self.ap_settings.udp_timeout)
+        except (AttributeError, KeyError):
+            # Some UDP value is missing, toss it all.
+            pass
```

Comment:
If we maintain a separate list of IWakeup implementations, each could have a 'check_hello' method that is called, to determine what impl is used. This keeps the logic out of websocket.py (where it can easily grow quite a bit if/when we add more wakeup schemes).